core/src/test/scala/integration/kafka/api/SaslSslEndToEndAuthorizationTest.scala

```diff
@@ -16,17 +16,79 @@
   */
 package kafka.api
 
+import scala.collection.immutable.List
+import scala.collection.JavaConverters._
+
+import org.junit.Test
+import java.util.Properties
+import java.io.File
 import kafka.server.KafkaConfig
+import kafka.utils.TestUtils
+import kafka.utils.JaasTestUtils
+import kafka.utils.JaasTestUtils.Krb5LoginModule
 import org.apache.kafka.common.protocol.SecurityProtocol
+import org.apache.kafka.common.KafkaException
+import org.apache.kafka.common.config.SaslConfigs
+import org.apache.kafka.common.errors.GroupAuthorizationException
 
 class SaslSslEndToEndAuthorizationTest extends EndToEndAuthorizationTest {
   override protected def securityProtocol = SecurityProtocol.SASL_SSL
-  override val clientPrincipal = "client"
-  override val kafkaPrincipal = "kafka"
+  override val clientPrincipal = JaasTestUtils.KafkaClientPrincipalUnqualifiedName
+  override val kafkaPrincipal = JaasTestUtils.KafkaServerPrincipalUnqualifiedName
+  
+  var clientKeytabFile: Option[File] = None
 
   // Configure brokers to require SSL client authentication in order to verify that SASL_SSL works correctly even if the
   // client doesn't have a keystore. We want to cover the scenario where a broker requires either SSL client
   // authentication or SASL authentication with SSL as the transport layer (but not both).
   serverConfig.put(KafkaConfig.SslClientAuthProp, "required")
 
+  // Use JAAS configuration properties for clients so that dynamic JAAS configuration is also tested by this set of tests
+  override protected def setJaasConfiguration(mode: SaslSetupMode, serverMechanisms: List[String], clientMechanisms: List[String],
```

Comment:
The same method could be used for both SaslSslEndToEndAuthorizationTest and SaslPlainSslEndToEndAuthorizationTest to avoid duplication?